lib/rules/color-function-notation/index.js

```diff
@@ -46,6 +60,13 @@ const rule = (primary, _secondaryOptions, context) => {
 
 				const { value, sourceIndex, sourceEndIndex, nodes } = node;
 
+				if (
+					ignoreWithVarInside &&
+					nodes.filter((child) => child.type === 'function' && child.value.toLowerCase() === 'var')
+						.length > 0
+				)
+					return;
```

Comment:
[suggest] I recommend using `.some()` over `.filter().length`. Also, using `{}` for `return` at a new line seems more readable.

```suggestion
				if (
					ignoreWithVarInside &&
					nodes.some((child) => child.type === 'function' && child.value.toLowerCase() === 'var')
				) {
					return;
				}
```

Reply:
Or, extracting a helper function might be more readable like this:

```js
if (ignoreWithVarInside && containsVariable(nodes)) return;

// ...

/**
 * @param {import('postcss-value-parser').Node[]} nodes
 */
function containsVariable(nodes) {
	return nodes.some(({ type, value }) => type === 'function' && value.toLowerCase() === 'var');
}
```